lib/spoom/deadcode/plugins/base.rb

```diff
@@ -159,26 +159,26 @@ def internal_on_define_accessor(indexer, definition)
         #
         # ~~~rb
         # class MyPlugin < Spoom::Deadcode::Plugins::Base
-        #   def on_define_class(indexer, definition)
-        #     definition.ignored! if definition.name == "Foo"
+        #   def on_define_class(symbol_def, definition)
+        #     definition.ignored! if symbol_def.name == "Foo"
         #   end
         # end
         # ~~~
-        sig { params(indexer: Indexer, definition: Definition).void }
-        def on_define_class(indexer, definition)
+        sig { params(symbol_def: Model::Class, definition: Definition).void }
```

Comment:
Should this be abstract?

Reply:
We don't want to crash when calling `on_define_class` to a plugin that do nothing with it, we just want it to no-op.

We _could_refactor this to have multiple interfaces:

```rb
module ClassPlugin
  interface!

  sig { abstract.void }
  def on_define_class; end
```

and then let the plugins cherry pick the interfaces they want to implement.

But in practice I just rely on YJIT to optimize this empty calls away.